packages/@jsii/kernel/lib/kernel.ts

```diff
@@ -107,14 +107,23 @@ export class Kernel {
 
     // Create the install directory (there may be several path components for @scoped/packages)
     fs.mkdirpSync(packageDir);
-    // untar the archive to its final location
-    tar.extract({
-      strict: true,
-      file: req.tarball,
-      cwd: packageDir,
-      strip: 1, // Removes the 'package/' path element from entries
-      sync: true,
-    });
+
+    // Force umask to have npm-install-like permissions
+    const originalUmask = process.umask(0o022);
+    try {
+      // untar the archive to its final location
+      tar.extract({
```

Comment:
You should use the built-in support in the [`tar`](https://www.npmjs.com/package/tar#tarcoptions-filelist-callback-alias-tarcreate) library.

One of these options should help:

* `portable` Omit metadata that is system-specific: ctime, atime, uid, gid, uname, gname, dev, ino, and nlink. Note that mtime is still included, because this is necessary for other time-based operations. Additionally, mode is set to a "reasonable default" for most unix systems, based on a umask value of 0o22.
* `mode` The mode to set on the created file archive

Reply:
I feel `portable` is the right one to use here.

Reply:
Those options are specific to the `create` operation and this is the `extract` operation.

The only "built-in" option available is the `umask` option to `tar.extract`, however that is "internal" (and not exposed in the `d.ts` which is why I **decided** against using that).

Reply:
Okay sorry my mistake.